packages/graph/hash_graph/lib/graph/src/types/schema/array.rs

```diff
@@ -8,86 +8,41 @@ enum ArrayTypeTag {
     Array,
 }
 
-#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct Array {
+pub struct Array<T> {
     r#type: ArrayTypeTag,
+    items: T,
     #[serde(skip_serializing_if = "Option::is_none")]
     min_items: Option<usize>,
     #[serde(skip_serializing_if = "Option::is_none")]
     max_items: Option<usize>,
 }
 
-impl Array {
+impl<T> Array<T> {
     #[must_use]
-    pub const fn new(min_items: Option<usize>, max_items: Option<usize>) -> Self {
+    pub const fn new(items: T, min_items: Option<usize>, max_items: Option<usize>) -> Self {
         Self {
             r#type: ArrayTypeTag::Array,
+            items,
             min_items,
             max_items,
         }
     }
 
-    #[must_use]
-    pub const fn min_items(&self) -> Option<usize> {
-        self.min_items
-    }
-
-    #[must_use]
-    pub const fn max_items(&self) -> Option<usize> {
-        self.max_items
-    }
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct Itemized<A, T> {
-    #[serde(flatten)]
-    array: A,
-    items: T,
-}
-
-impl<A, T> Itemized<A, T> {
-    #[must_use]
-    pub const fn new(array: A, items: T) -> Self {
-        Self { array, items }
-    }
-
-    #[must_use]
-    pub const fn array(&self) -> &A {
-        &self.array
-    }
-
     #[must_use]
     pub const fn items(&self) -> &T {
         &self.items
     }
-}
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct MaybeOrdered<A> {
```

Comment:
As we only have `Array` and `MaybeOrderedArray`, this should probably live next to each other again.

Reply:
I didn't know, that `ordered` is not a term in JSON Schema but added from us.